### ISSUE TITLE
If no doi, study shows as [no year] even if a year is listed

### ISSUE THREAD

**Issue by snacktavish** [OpenTreeOfLife/opentree#669]
e.g. https://tree.opentreeoflife.org/curator/study/view/ot_299
I'm not sure if this is the behaviour we want or not, but I found it a bit unintuitive.

**Comment by jimallman** [comment]
I see this (missing year if no DOI) in the study list, but not in the single-study curation page. Definitely a bug!

**Comment by jimallman** [comment]
Ah, now I see that the year is also missing in the main title of the study-curation page. I suspect the same bug is behind both.

@snacktavish, your example study **ot_299** has apparently been deleted, so I'm investigating this using [study **ot_301** on **devtree**](https://tree.opentreeoflife.org/curator/study/edit/ot_301), which shows similar behavior.

<img width="634" alt="screen shot 2015-07-10 at 12 11 42 pm" src="https://cloud.githubusercontent.com/assets/446375/8623052/13d54346-26fd-11e5-9f5a-37a753991a23.png">

In this case, we're trying to show a "compact" reference (Smith, 2001) built from the full reference text. Currently this disregards the stored publication year and tries to use only the reference text itself, which is very incomplete. So the [no year] is symptomatic of bad/incomplete reference text, and arguably not a bug after all. Note that the red text is complaining about this real problem in the data, a mismatch between the study's reference text and its year of publication. 

I could grab that separate year value and show it, or leave this in place as an additional warning that the data needs attention. Thoughts?

**Comment by snacktavish** [comment]
ah whoops, I deleted the original one. Yeah, I'm not sure if it is a bug or not. The curator had gotten a nexson validation error (below) when she tried to upload a tree, that I thought could be related to lack of study year? But then it was not reproducible at all, and trees upload fine to studies not showing a year, so I'm not sure what was going on.

  {"description": "ot_validation failed: {\"annotationEvent\": {\"@preserve\": false, \"@description\": \"Open Tree NexSON validation\", \"@passedChecks\": false, \"@wasAssociatedWithAgentId\": \"peyotl-validator\", \"message\": [{\"@severity\": \"ERROR\", \"@code\": \"INCORRECT_VALUE_TYPE\", \"data\": [{\"expected\": \"integer\", \"type\": \"NoneType\", \"key\": \"^ot:studyYear\"}], \"refersTo\": {\"@idref\": \"study\", \"@top\": \"nexml\"}}], \"@id\": \"peyotl-validator-event\"}, \"agent\": {\"@des
cription\": \"Open Tree NexSON validation\", \"@name\": \"api.opentreeoflife.org/validate\", \"@version\": \"0.0.4a\", \"@url\": \"https://github.com/OpenTreeOfLife/peyotl\", \"invocation\": {\"otherProperty\": [{\"name\": \"pythonVersion\", \"value\": \"2.7.9\"}, {\"name\": \"pythonImplementation\", \"value\": \"CPython\"}], \"commandLine\": []}, \"@id\": \"peyotl-validator\"}}", "error": 1}

**Comment by jimallman** [comment]
> The curator had gotten a nexson validation error (below) when she tried to upload a tree, that I thought could be related to lack of study year?

Yes, that's exactly what's indicated. It looks like a new study with no year set, which caused the validation error. I'm surprised, since we decided the validator should be _very_ forgiving of incomplete studies to encourage new curators.

Hm, a quick test shows that I can't save a study with no Study Year, but even one digit here will satisfy the validator. With absolutely no other data (other than the initial DOI and CC-0 waiver), I was [able to save this study](http://devtree.opentreeoflife.org/curator/study/view/tt_46) on devtree.

@mtholder, @snacktavish: Is there any reason we shouldn't relax the Nexson validator to allow either an integer or None/null for the study year? It looks like [this is the line to change](https://github.com/OpenTreeOfLife/peyotl/blob/aafaaf757d025024dce83c1d5b54cad6f86fd440/peyotl/nexson_validation/schema.py#L334), but I don't see an existing data-type in `_VT` that would accept either an integer **or** None.

Should we add something like `_VT.INT_OR_NONE`, or `_VT.OPTIONAL_INT`? Or should we just move the `ot:studyYear` check from the [expected](https://github.com/OpenTreeOfLife/peyotl/blob/aafaaf757d025024dce83c1d5b54cad6f86fd440/peyotl/nexson_validation/schema.py#L331) to the [allowed](https://github.com/OpenTreeOfLife/peyotl/blob/aafaaf757d025024dce83c1d5b54cad6f86fd440/peyotl/nexson_validation/schema.py#L324) list of element checks? Or add an `allowed_meta` collection?

**Comment by jimallman** [comment]
Reminder: The several different types of element checks are briefly [described in the `_SchemaFragment` class](https://github.com/OpenTreeOfLife/peyotl/blob/aafaaf757d025024dce83c1d5b54cad6f86fd440/peyotl/nexson_validation/schema.py#L36-L42):

-        `required` dict of key to _VT type code. Error is not present.
-        `expected` dict of key to _VT type code. Warnings is not present.
-        `allowed` dict of None ignored elements #@TODO could be a set.
-        `required_meta` dict of meta key to _VT type code. Error is not present.
-        `expected_meta` dict of meta key to _VT type code. Warnings is not present.
-        `type_checked_meta` dict of meta key to _VT type code. Error if wrong type, no warning if -absent.
-        `using_hbf_meta` True for >=1.0.0, False for badgerfish.
